homeassistant/components/climate/device_action.py

```diff
@@ -59,14 +59,16 @@
                 CONF_TYPE: "set_hvac_mode",
             }
         )
-        actions.append(
-            {
-                CONF_DEVICE_ID: device_id,
-                CONF_DOMAIN: DOMAIN,
-                CONF_ENTITY_ID: entry.entity_id,
-                CONF_TYPE: "set_preset_mode",
-            }
-        )
+
+        if state and const.ATTR_PRESET_MODES in state.attributes:
```

Comment:
We should use the supported feature flag.

Reply:
I agree, but thats  the way it is implemented in device_trigger and device_condition.
Should i change it there too.

Reply:
I modified device_condition too